### PR TITLE
[WIP] Produce ansible_version related to venv

### PR DESCRIPTION
**Pull request by AlanCoding**
This intentionally introduces a conflict with https://github.com/ansible/awx/pull/3041, because this logic is central to the implementation of inventory plugins, and I would rather get ahead of the conflicts.

We were previously producing this parameter that was passed around tasks.py that gave the Ansible version, it just wasn't really the version that particular job run used. Inventory plugins will have to use the version over and over again, because not all plugins works in all Ansible versions.

### CODE REVIEW

**Review by ryanpetrello — APPROVED** [review]

**Comment by AlanCoding** [comment]
Going to wait on @rooftopcellist later today before merge

File: awx/main/utils/common.py
Comment (AlanCoding) [inline awx/main/utils/common.py:186]: This isn't a good practice here.

Since this method is memoized, it shouldn't take any argument. That only begs for bugs. I say this should take no argument, and call `_get_ansible_version('ansible')`

Comment (jpic) [reply awx/main/utils/common.py]: Why does this need an argument at all ?

Shouldn't the code that's in [BaseTask.get_ansible_version()](https://github.com/ansible/awx/blob/6306255e43fcd04e8cc16532a303504f5ae0f5c0/awx/main/tasks.py#L692-L697), which does *not* use `self`, be defined at least in this module by default ?

**Comment by ryanpetrello** [comment]
@AlanCoding is this PR ready to be closed given https://github.com/ansible/awx/pull/3266/files#diff-1b22494f51df57adb72248c06506a997L157

**Comment by AlanCoding** [comment]
yeah

**Pull request closed without merging**
